frontend/src/parser/crCalculator.ts

```diff
@@ -0,0 +1,110 @@
+import type { FeatureEntity, Saves } from "../../../shared"
+import { parseDescIntoAutomation } from "./utils"
+
+export function scrapeFeatures(features: FeatureEntity[] = []): number[]{
+    let bonuses: number[] = []
+    let damages: number[] = []
+    let saveDC: number[] = []
+
+    for (const feature of features){
+        let automation: object | null
+
+        if (feature.automation){
+            automation = feature.automation
+        } else {
+            automation = parseDescIntoAutomation(feature.description, feature.name ,1)[0]
+        }
+
+        const bonus = automation ?  findAttackBonus(automation) : undefined
+        const damage = automation ? findAttackDamage(automation) : undefined
+        const dc = automation ? findDC(automation): undefined
+
+        if (bonus && !isNaN(bonus)){
+            bonuses.push(bonus)
+        }
+
+        if (damage && !isNaN(damage)){
+            damages.push(damage)
+        }
+
+        if (dc && !isNaN(dc)){
+            saveDC.push(dc)
+        }
+    }
+
+    const averageAttackBonus = averageValue(bonuses)
+    const averageDPR = averageValue(bonuses)
+    const averageDC = averageValue(saveDC)
+
+    return [averageAttackBonus, averageDPR, averageDC]
+}
+
+export function countProficientSaves(saves: Saves): number {
+    let totalCount = 0
+
+    for (const [key, save] of Object.entries(saves)){
```

Comment:
using entries here is unnecessary. use `for (const save of saves)` instead (like done above)

Reply:
saves by itself isn't iterable though. Doing (const save of saves) throws a ts error of Type 'Saves' must have a '[Symbol.iterator]()' method that returns an iterator.ts(2488)

Reply:
Oh whoops. Then use `for (const save of Object.values(saves))`